Doc/library/security_warnings.rst

```diff
@@ -0,0 +1,14 @@
+.. _security-warnings:
+
+.. index:: single: security considerations
+
+Security Considerations
+=======================
+
+The following modules have specific security considerations:
+
+* :mod:`ssl` : :ref:`SSL/TLS security considerations <ssl-security>`
+* :mod:`cgi` : :ref:`CGI security considerations <cgi-security>`
+* :mod:`pickle` : :ref:`Restricting globals in pickle <pickle-restrict>`
+* :mod:`subprocess` : :ref:`Subprocess security considerations <subprocess-security>`
+* :mod:`xml` : :ref:`XML vulnerabilities <xml-vulnerabilities>`
```

Comment:
There are also:
* https://docs.python.org/dev/library/http.server.html
* logging config uses eval
* hashlib: usedforsecurity=True
* multiprocessing recv uses pickle
* random: "The pseudo-random generators of this module should not be used for security purposes."
* zipfile ZIP bomb: https://docs.python.org/dev/library/zipfile.html#resources-limitations
* os.popen() is vulnerable to shell injection

Maybe this documentation page could be more ambitious and give more hints about security? Examples:

* Discourage the usage of eval(),
* Warn about bugs in Python itself (ex: ast.literal_eval doesn't protect against DoS and bugs in Python parser).
* Windows DLL injection
* sys.path: PYTHONPATH, -I, -E, -s and -S command line options
* Security of embedded Python
* PyPI Supplain Chain

You may even mention tools dedicated to search for Python vulnerabilities, like: https://pypi.org/project/bandit/

See also my notes:

* https://python-security.readthedocs.io/ssl.html
* https://python-security.readthedocs.io/security.html
* https://python-security.readthedocs.io/packages.html
* https://python-security.readthedocs.io/vulnerabilities.html